File: sdk-examples/python/hello.py
"""Hello World — Minimal EdgeCrab Python SDK Example.

Usage:
    python hello.py
"""

from edgecrab import Agent


def main():
    # Create an agent with a model string
    agent = Agent("anthropic/claude-sonnet-4", quiet_mode=True)

    # Send a message (sync)
    reply = agent.chat_sync("What is EdgeCrab? Answer in one sentence.")
    print(f"Agent: {reply}")


if __name__ == "__main__":
    main()

File: sdk-examples/rust/hello.rs
//! # Hello World — Minimal EdgeCrab SDK Example
//!
//! Send a single message and print the reply.
//!
//! ```bash
//! cargo run --example hello
//! ```

use edgecrab_sdk::prelude::*;

#[tokio::main]
async fn main() -> Result<(), SdkError> {
    // Create an agent with the default model
    let agent = SdkAgent::new("anthropic/claude-sonnet-4")?;

    // Send a message
    let reply = agent.chat("What is EdgeCrab? Answer in one sentence.").await?;
    println!("Agent: {reply}");

    Ok(())
}
